cpp/daal/include/algorithms/k_nearest_neighbors/kdtree_knn_classification_predict.h

```diff
@@ -274,8 +274,145 @@ class Batch : public classifier::prediction::Batch
 /** @} */
 } // namespace interface2
 
-using interface2::BatchContainer;
-using interface2::Batch;
+namespace interface3
+{
+/**
+ * @defgroup kdtree_knn_classification_prediction_batch Batch
+ * @ingroup kdtree_knn_classification_prediction
+ * @{
+ */
+
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__KDTREE_KNN_CLASSIFICATION__PREDICTION__BATCHCONTAINER"></a>
+ *  \brief Class containing computation methods for KD-tree based kNN model-based prediction
+ */
+template <typename algorithmFPType, Method method, CpuType cpu>
+class BatchContainer : public PredictionContainerIface
+{
+public:
+    /**
+     * Constructs a container for KD-tree based kNN model-based prediction with a specified environment
+     * \param[in] daalEnv   Environment object
+     */
+    BatchContainer(daal::services::Environment::env * daalEnv);
+
+    ~BatchContainer();
+
+    /**
+     *  Computes the result of KD-tree based kNN model-based prediction
+     */
+    services::Status compute() DAAL_C11_OVERRIDE;
+};
+
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__KDTREE_KNN_CLASSIFICATION__PREDICTION__BATCH"></a>
+ * \brief Provides methods to run implementations of the KD-tree based kNN model-based prediction
+ * <!-- \n<a href="DAAL-REF-KNN-ALGORITHM">kNN algorithm description and usage models</a> -->
+ *
+ * \tparam algorithmFPType  Data type to use in intermediate computations for KD-tree based kNN model-based prediction
+ *                          in the batch processing mode, double or float
+ * \tparam method           Computation method in the batch processing mode, \ref Method
+ *
+ * \par Enumerations
+ *      - \ref Method  Computation methods for KD-tree based kNN model-based prediction
+ *
+ * \par References
+ *      - \ref kdtree_knn_classification::interface1::Model "kdtree_knn_classification::Model" class
+ *      - \ref training::interface1::Batch "training::Batch" class
+ */
+template <typename algorithmFPType = DAAL_ALGORITHM_FP_TYPE, Method method = defaultDense>
+class Batch : public classifier::prediction::Batch
+{
+public:
+    typedef classifier::prediction::Batch super;
+
+    typedef algorithms::kdtree_knn_classification::prediction::Input InputType;
+    typedef algorithms::kdtree_knn_classification::Parameter ParameterType;
+    typedef algorithms::kdtree_knn_classification::prediction::Result ResultType;
+
+    InputType input;         /*!< %Input data structure */
+    ParameterType parameter; /*!< \ref kdtree_knn_classification::interface3::Parameter "Parameters" of prediction */
```

Comment:
why we have parameter as a value here?
seems `parameter()` method should be added.

Reply:
Sure. I plan to add it with another modifications for test system by another PR